lisa/tools/fio.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from decimal import Decimal
+from typing import Any, Dict, List, Type, cast
+
+from lisa.executable import Tool
+from lisa.notifier import FIOResult, FIOResultsMessage
+from lisa.operating_system import Debian, Posix, Redhat, Suse
+from lisa.util import LisaException
+
+from .git import Git
+
+
+class Fio(Tool):
+    fio_repo = "https://github.com/axboe/fio/"
+    branch = "fio-3.29"
+    _result_pattern = re.compile(
+        r"([\w\W]*?)IOPS=(?P<iops>.+?)," r"([\w\W]*?).* lat.*avg=(?P<latency>.+?),",
+        re.M,
+    )
+
+    @property
+    def command(self) -> str:
+        return "fio"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git]
```

Comment:
Git is not a hard dependency, because the tool may be installed from packages. So don't add git here.

Reply:
updated